### PR TITLE
chore(rules): add type to QueueRules

### PR DESCRIPTION
**Pull request by sileht**
Change-Id: Ida624bfab115548cf2c0b5dc8fa31aedda62cb78

### CODE REVIEW

**Comment by sileht** [comment]
This pull request is part of a stack:
1. fix(queue): preserve order when train is skrink ([#2960](https://github.com/Mergifyio/mergify-engine/pull/2960))
1. chore(rules): add type to QueueRules ([#2961](https://github.com/Mergifyio/mergify-engine/pull/2961))
1. feat(queue): sync configuration of queued PR ([#2962](https://github.com/Mergifyio/mergify-engine/pull/2962))

File: mergify_engine/rules/__init__.py [hunk 1]
```diff
@@ -550,23 +550,23 @@ async def get_pull_request_rule(self, ctxt: context.Context) -> RulesEvaluator:
...
         return len(self.rules)
 
-    def __post_init__(self):
-        names = set()
+    def __post_init__(self) -> None:
+        names: typing.Set[QueueName] = set()
         for i, rule in enumerate(reversed(self.rules)):
             rule.config["priority"] = i
-            if rule.name is names:
+            if rule.name in names:
```
Comment (jd) [inline mergify_engine/rules/__init__.py:569]: sounds like we didn't have a test for this and your commit message is different than what it says?

**Review by jd — DISMISSED** [review]

File: mergify_engine/rules/__init__.py (hunk 1 above)
Comment (jd) [reply mergify_engine/rules/__init__.py:569]: @sileht ?

Comment (sileht) [reply mergify_engine/rules/__init__.py:569]: https://github.com/Mergifyio/mergify-engine/pull/2964

**Review by jd — APPROVED** [review]

**Pull request merged**